pixeltable/functions/voyageai.py

```diff
@@ -0,0 +1,161 @@
+            Most models only support a single default dimension. Models `voyage-3-large`, `voyage-3.5`,
+            `voyage-3.5-lite`, and `voyage-code-3` support: 256, 512, 1024 (default), and 2048.
+        output_dtype: The data type for the embeddings to be returned. Options: `float`, `int8`, `uint8`,
+            `binary`, `ubinary`. Only `float` is currently supported in Pixeltable.
+
+    Returns:
+        An array representing the application of the given embedding to `input`.
+
+    Examples:
+        Add a computed column that applies the model `voyage-3.5` to an existing
+        Pixeltable column `tbl.text` of the table `tbl`:
+
+        >>> tbl.add_computed_column(embed=embeddings(tbl.text, model='voyage-3.5', input_type='document'))
+
+        Add an embedding index to an existing column `text`, using the model `voyage-3.5`:
+
+        >>> tbl.add_embedding_index('text', text_embed=embeddings.using(model='voyage-3.5'))
+    """
+    cl = _voyageai_client()
+
+    # Build kwargs for the API call
+    kwargs = {}
+    if input_type is not None:
+        kwargs['input_type'] = input_type
+    if truncation is not None:
+        kwargs['truncation'] = truncation
+    if output_dimension is not None:
+        kwargs['output_dimension'] = output_dimension
+    if output_dtype is not None:
+        kwargs['output_dtype'] = output_dtype
+
+    result = await cl.embed(texts=input, model=model, **kwargs)
+    return [np.array(emb, dtype=np.float64) for emb in result.embeddings]
+
+
+@pxt.udf(resource_pool='request-rate:voyageai')
+async def rerank(
+    query: str, documents: list[str], *, model: str, top_k: int | None = None, truncation: bool = True
+) -> dict:
+    """
+    Reranks documents based on their relevance to a query.
+
+    Equivalent to the Voyage AI `rerank` API endpoint.
+    For additional details, see: <https://docs.voyageai.com/docs/reranker>
+
+    Request throttling:
+    Applies the rate limit set in the config (section `voyageai`, key `rate_limit`). If no rate
+    limit is configured, uses a default of 600 RPM.
+
+    __Requirements:__
+
+    - `pip install voyageai`
+
+    Args:
+        query: The query as a string.
+        documents: The documents to be reranked as a list of strings.
+        model: The model to use for reranking. Recommended options: `rerank-2.5`, `rerank-2.5-lite`.
+        top_k: The number of most relevant documents to return. If not specified, all documents
+            will be reranked and returned.
+        truncation: Whether to truncate the input to satisfy context length limits. Defaults to `True`.
+
+    Returns:
+        A dictionary containing:
+        - `results`: List of reranking results with `index`, `document`, and `relevance_score`
+        - `total_tokens`: The total number of tokens used
+
+    Examples:
+        Rerank a list of documents based on relevance to a query:
+
+        >>> tbl = pxt.create_table('docs', {'query': pxt.String, 'docs': pxt.Json})
```

Comment:
I would use a more appropriate example: instead of 'docs', add a computed column with a @query udf that does a similarity query and selects only the text pieces.

That's a typical scenario where you'd use reranking.